aragon/court-dashboard
# Mock data

This is an initial implementation to test the court without the burdens to setup everything locally or have a way to test different dispute phases as well as final rounds. 

TODO: 

- [ ] Add claimed rewards data
- [ ] Add total jurors stat
- [ ] General Clean up code
- [ ] Add comments

This implementation attempts to keep a consistent data across the dashboard so some contract logic had to be implemented :D 


### Considerations

- Currently there are about 11 disputes created, each one of them with a different state or different result. 

- There are 5 harcoded accounts that have participation in the court and these are the first 5 accounts from the default mnemonic that we use locally
 ```
myth like bonus scare over problem client lizard pioneer submit female collect
```

The first 3 accounts are jurors, the 4th is the appealer and the 5th the confirm appealer.

- Draft: For each round created we created drafts randomly between these 3 jurors, so on every page refresh you could see different results (jurors drafted for different disputes, dispute current ruling distribution changed, etc). What remains always unchanged are the dispute final results and  previous round results (this will always be outcome: Blocked Action, appeal: Allow Action, Confirm appeal: Blocked action).

- Since we get some data from the contracts itself,  this data is not available (e.g Subscription rewards, Total ANT staked, appeal deposits amount, etc)

- Actions are not possible here, it's read only for now.

### How to test

You can run it locally by `npm run start:local:mock`

I recommend using Rinkeby or Mainnet since running locally requires to run Ganache and possibly also ipfs to see the disputes content (we are using the recent App mining dispute metadata for all disputes)

So you can do 

```
npm run start:rinkeby:mock
#or
npm run start:mainnet:mock
```
